src/main/java/ezschedule/model/event/Event.java

```diff
@@ -61,21 +68,50 @@ public boolean equals(Object other) {
 
         Event otherEvent = (Event) other;
         return otherEvent.getName().equals(getName())
-            && otherEvent.getDate().equals(getDate())
-            && otherEvent.getStartTime().equals(getStartTime())
-            && otherEvent.getEndTime().equals(getEndTime());
+                && otherEvent.getDate().equals(getDate())
+                && otherEvent.getStartTime().equals(getStartTime())
+                && otherEvent.getEndTime().equals(getEndTime());
     }
 
     @Override
     public String toString() {
         final StringBuilder sb = new StringBuilder();
         sb.append(getName())
-            .append("\nDate: ")
-            .append(getDate())
-            .append("\nStart Time: ")
-            .append(getStartTime())
-            .append("\nEnd End: ")
-            .append(getEndTime());
+                .append("\nDate: ")
+                .append(getDate())
+                .append("\nStart Time: ")
+                .append(getStartTime())
+                .append("\nEnd End: ")
+                .append(getEndTime());
         return sb.toString();
     }
+
+    private boolean isEqualDate(Event otherEvent) {
+        return this.getDate().equals(otherEvent.getDate());
+    }
+
+    private boolean isTimeOverlap(Event otherEvent) {
```

Comment:
Would it be an issue for the following?
- otherEvent s/10:00 e/12:00
- currEvent s/09:00 e/11:00

Reply:
No. if the currEvent is the alreadt existing event, the otherEvent will not be added as it overlaps in time